### DISCUSSION TITLE
How to Access Response Object in express-zod-api Handler to Set Cookies

### DISCUSSION THREAD

**Discussion by yuuri111** [RobinTail/express-zod-api#1754] [Q&A] [answered]
Hello,

I am currently working with express-zod-api and trying to set cookies directly in the response object within a handler. However, I noticed that the response object is not directly available in the handler's parameters as shown below:

```ts
export const test = defaultEndpointsFactory.build({
  method: "post",
  tag: "test",
  description: "test",
  input: z.object({}),
  output: z.object({
    test: z.string(),
  }),
  handler: async ({ input }) => {
    // I need to use response.set here
    // Example:
    // response.set("Set-Cookie", `test=${test}; HttpOnly; Secure; Path=/; Max-Age=3600`);
    
    const test = "value";
    return {
      test
    };
  }
});
```

Could you advise on how I can modify the handler or the factory setup so that I can access the response object to use methods like response.set for setting cookies? I am looking to manage HTTP headers directly, particularly for setting cookies.

Thank you for your assistance.

**Comment by RobinTail** [comment]
Hello @yuuri111 

Here are existing topics for you on that matter:
- https://github.com/RobinTail/express-zod-api/discussions/710
- https://github.com/RobinTail/express-zod-api/discussions/925

In short:
- The `request` is only available for middlewares and Result Handlers.
- You can either:
  - set the cookie there,
  - or expose "cookie setter" or whole `request` from a custom middleware, to access it from `options` in endpoint's handler.

Please let me know if more information required.

**Reply by RobinTail** [reply]
```ts
const cookieMw = createMiddleware({
  input: z.object({}),
  middleware: async ({ request }) => ({
    setCookie: () => {}
  })
});

const endpoint = defaultEndpointsFactory
  .addMiddleware(cookieMw)
  .build({
    handler: async ({ options }) => { options.setCookie() }
  });
```

@yuuri111

**Reply by yuuri111** [reply]
Thank you so much!

**Comment by RobinTail** [comment, marked as answer]
First-class cookie support added in v28.1.0, @yuuri111 
https://ez.robintail.cz/v28.1.0/cookies
